ArdusatSDK.cpp

```diff
@@ -98,211 +108,51 @@ void _writeBeginError(const char sensorName[] PROGMEM)
-  lsm303_getAccel(&(output.x), &(output.y), &(output.z));
-}
-
-/*
- * Magnetic Field
- */
-boolean beginMagneticSensor() {
-  return start_sensor_or_err(magnetic_sensor_name, lsm303_mag_init);
-}
-
-void readMagnetic(magnetic_t & output) {
-  output.header.unit = DATA_UNIT_MICROTESLA;
-  output.header.timestamp = millis();
-
-  output.header.sensor_id = SENSORID_ADAFRUIT9DOFIMU;
-  lsm303_getMag(&(output.x), &(output.y), &(output.z));
-}
-
-/*
- * Gyroscope
- */
-boolean beginGyroSensor() {
-  return start_sensor_or_err(gyro_sensor_name, l3gd20h_init);
-}
-
-void readGyro(gyro_t & output) {
-  output.header.unit = DATA_UNIT_RADIAN_PER_SECOND;
-  output.header.timestamp = millis();
-
-  output.header.sensor_id = SENSORID_ADAFRUIT9DOFIMU;
-  l3gd20h_getOrientation(&(output.x), &(output.y), &(output.z));
-}
-
-/*
- * Barometric Pressure
  */
-boolean beginBarometricPressureSensor() {
-  return start_sensor_or_err(pressure_sensor_name, bmp180_init);
-}
+void _writeUnavailableSensorError(const char sensorName[] PROGMEM, const char hardwareBuild[] PROGMEM) {
+  char err_msg[55];
+  char sensor[50];
+  char hardware[15];
 
-void readBarometricPressure(pressure_t & output) {
-  output.header.unit = DATA_UNIT_HECTOPASCAL;
-  output.header.timestamp = millis();
+  strcpy_P(err_msg, unavailable_on_spaceboard_error_msg);
+  strcpy_P(sensor, sensorName);
+  strcpy_P(hardware, hardwareBuild);
 
-  output.header.sensor_id = SENSORID_BMP180;
-  bmp180_getPressure(&output.pressure);
+  //Make SURE sensorName isn't too long for the output buffer!!!
+  _resetOutBuf();
+  sprintf(_getOutBuf(), err_msg, sensor, hardware);
+  Serial.println(_getOutBuf());
 }
 
 /**
- * Calculate the orientation (pitch, roll, heading) from raw magnetometer and
- * accelerometer readings.
- *
- * Calculation based on Adafruit 9DOF library
- * (https://github.com/adafruit/Adafruit_9DOF/)
+ * Accepts initialization functions as a parameter. Tries to initialize functions
+ * but prints error messages if fails.
  *
- * @param accel Acceleration reading to use in calculation
- * @param mag Magnetometer reading to use in calculation
- * @param orient Orientation structure to save calculated orientation in
+ * @param sensorName name of sensor that failed.
+ * @param init_func function to try and initialize sensor
  */
-void calculateOrientation(const acceleration_t & accel, const magnetic_t & mag,
-			  orientation_t & orient)
-{
-  const float PI_F = 3.141592653F;
-
-  // Roll is rotation around x-axis (-180 <= roll <= 180)
-  // Positive roll is clockwise rotation wrt positive x axis
-  orient.roll = (float) atan2(accel.y, accel.z);
-
-  // Pitch is rotation around y-axis (-180 <= pitch <= 180)
-  // Positive pitch is clockwise rotation wrt positive y axis
-  if (accel.y * sin(orient.roll) + accel.z * cos(orient.roll) == 0) {
-    orient.pitch = accel.x > 0 ? (PI_F / 2) : (-PI_F / 2);
+boolean start_sensor_or_err(const char sensorName[] PROGMEM, boolean initialized) {
```

Comment:
Now that we're not calling the init function here, `start_sensor_or_err` isn't a very descriptive name for this function. I'd probably just put the onus on each init function to call `_writeBeginError` directly and avoid the extraneous function call altogether.